src/Filter/AbstractDateFilter.php

```diff
@@ -172,6 +126,6 @@ protected function getOperator($type)
             DateOperatorType::TYPE_LESS_THAN => 'lt',
         ];
 
-        return $choices[(int) $type];
+        return $choices[$type];
```

Comment:
Maybe we should check that the passed offset is defined and throw an exception otherwise.